Source/WebCore/style/FilterOperationsBuilder.cpp

```diff
@@ -196,6 +199,20 @@ std::optional<FilterOperations> createFilterOperations(const Document& document,
     return operations;
 }
 
+std::optional<FilterOperations> createFilterOperations(const Document& document, RenderStyle& style, const String& string, CSSParserMode mode)
```

Comment:
If this can be called on a Worker thread, using Document is not valid.

Reply:
But it is called from `CanvasRenderingContext2D` which is created for the `HTMLCanvasElement`. The `filter` API is not implemented for the `OffscreenCanvas` yet.

Reply:
This is fixed by passing `FilterParserContext` to `createFilterOperations()`. This class will take a `Document` only when parsing the filter for `CSS` and  `<canvas>` element. For `OffscreenCanvas` it will fall back to the default `FilterParserContext`.